docker/Dockerfile

```diff
@@ -6,26 +6,81 @@
 #    docker build -f docker/Dockerfile .
 #
 
+ARG PYTHON_VERSION=3.11
+
+###
+### Stage 0: generate requirements.txt
+###
+# We hardcode the use of Debian bookworm here because this could change upstream.
+FROM docker.io/library/python:${PYTHON_VERSION}-slim-bookworm as requirements
+
+# We install poetry in its own build stage to avoid its dependencies conflicting with
+# sygnal's dependencies.
+RUN --mount=type=cache,target=/root/.cache/pip \
+  pip install --user "poetry==1.3.2"
```

Comment:
That's quite an old version of poetry to pin to? It's now at 1.8.3

Reply:
I just grabbed the same version from synapse. I'll bump it up here since I don't think we have any of the same constraints.